src/client-side-encryption/stateMachine.ts

```diff
@@ -272,10 +336,10 @@ class StateMachine {
         reject(new MongoCryptError('KMS request timed out'));
       }
 
-      function onerror(err) {
+      function onerror(err: Error) {
         destroySockets();
-        const mcError = new MongoCryptError('KMS request failed');
-        mcError.originalError = err;
+        // TODO: make note of this
+        const mcError = new MongoCryptError('KMS request failed', { cause: err });
```

Comment:
This is a behavior change.  We now have access to the `cause` property - may as well make proper use of it.

Reply:
I assume this PR comment is the todo: `// TODO: make note of this`?